### PR TITLE
Fixed quote to heading transformation.

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
## Description
Fixes: https://github.com/WordPress/gutenberg/issues/5052

This PR corrects the logic of quote to heading transform.
The logic is now the following:
If no citation and no content exist an empty heading is created.
If citation exists but content does not exist a heading is created with citation as the content.
In other citations the first paragraph of the quote is transformed into and heading and new quote is created equal to the original but with the first paragraph removed. If the quote just contains one paragraph the new quote will contain just the citation.

## How Has This Been Tested?
Use transformations from quote to heading and verify the logic specified above is working correctly.

### CODE REVIEW

**Comment by youknowriad** [comment]
I had a JavaScript error by doing `quote => heading => quote`. Is it related? can we fix it as well?

**Comment by jorgefilipecosta** [comment]
>  I had a JavaScript error by doing quote => heading => quote. Is it related? can we fix it as well?

I was able to replicate the problem 👍 , it should be fixed now.

File: blocks/library/quote/index.js [hunk 1]
```diff
@@ -122,32 +122,40 @@ export const settings = {
...
-						const quote = createBlock( 'core/quote', {
-							...attrs,
-							citation,
-							value: Array.isArray( value ) ?
-								value.slice( 1 ) :
-								[],
-						} );
 
-						return [ text, quote ];
+					const headingContent = castArray( isString( get( value, [ 0, 'children' ] ) ) ?
```
Comment (youknowriad) [inline blocks/library/quote/index.js]: should we extract this value, [ 0, 'children' ]  to a variable? it seems used three times here?

**Review by youknowriad — APPROVED** [review]
I feel like we fixed this bug several times, some testing for these interactions would be good.

**Pull request merged**